src/Components/Components/src/Environment/ComponentEnvironment.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.Services;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Environment
+{
+    /// <summary>
+    /// Provides information an environment specific features about the environment the components are running on.
+    /// </summary>
+    public class ComponentEnvironment
```

Comment:
We might want to extend this with some sort of FeatureCollection like functionality so that different environments can save here their own custom abstractions. Like for example the Circuit on the case of the Remote rendering scenario.

Reply:
We have DI for that though. I don't think that it's needs to be that complicated

Reply:
That's a fair point.